src/Native/Unix/configure.cmake

```diff
@@ -321,6 +321,85 @@ check_function_exists(
     mach_timebase_info
     HAVE_MACH_TIMEBASE_INFO)
 
+set (PREVIOUS_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+set (CMAKE_REQUIRED_FLAGS "-Werror -Wsign-conversion")
+
+check_cxx_source_compiles(
+    "
+    #include <sys/socket.h>
+
+    int main()
+    {
+        int fd;
+        sockaddr* addr;
+        socklen_t addrLen;
+
+        int err = bind(fd, addr, addrLen);
+        return 0;
+    }
+    "
+    BIND_ADDRLEN_UNSIGNED
+)
+
+check_cxx_source_compiles(
+    "
+    #include <sys/socket.h>
+
+    int main()
+    {
+        int fd;
+        sockaddr* addr;
+        int32_t addrLen;
+
+        int err = bind(fd, addr, addrLen);
+        return 0;
+    }
+    "
+    BIND_ADDRLEN_SIGNED
+)
+
+if(NOT BIND_ADDRLEN_UNSIGNED AND NOT BIND_ADDRLEN_SIGNED)
+    message(FATAL_ERROR "The addrlen parameter in bind() must be a signed int or an unsigned int.")
+endif()
```

Comment:
That's correct, although you can say that about a lot of checks that take place in the CMake files. For example, you could argue we don't need to check for dependencies in CMake files because the code would fail to compile, anyway.

When porting to new platforms, my personal preference is to have the build process fail early. That way, you get warned upfront and there's usually a bit more context.

That said, happy to remove it if you think it's overkill.